Multiple OFF regions in `makers.make_theta_squared_table`

fabiopintore opened the pull request:
This PR includes a new feature in `~gammapy.makers.utils.make_theta_squared_table`. It will allow the users to choose the number of OFF regions, currently limited to only one (the mirror position). The implementation gives more flexibility to analysis of the theta2 plots.
Tests are also included in the PR.

The following example shows the new feature. For a given set of observations and target positions:

```
rad_axis = MapAxis.from_bounds(0, 0.5, nbin=25, unit="deg2", interp="lin")

off_regions_number = 5
theta2_dist = make_theta_squared_table(observations=observations, 
                                                                       theta_squared_axis=rad_axis, 
                                                                       position=target_position, 
                                                                       position_off=None,
                                                                       off_regions_number = off_regions_number,
                                       )
```

The OFF regions will be five, evenly spaced around the pointing position as in figure:

<img width="387" height="351" alt="Screenshot 2025-11-11 alle 08 28 20" src="https://github.com/user-attachments/assets/78c28fd9-798e-46ed-a4f1-f2f64fe1173d" />

and the theta2 plot for the OFF will take into account that `alpha` is now 0.2.

The `ReflectedRegionsFinder` approach cannot be adopted here as it limits the ON/OFF region size to the maximum angular separation to the pointing direction. However, a theta2 plot should be more flexible allowing for an unlimited exploration of the parameter space.

maxnoe commented:
Also, I think you've re-implemented the `WobbleRegionsFinder` here, you could just use it: https://docs.gammapy.org/2.0/api/gammapy.makers.WobbleRegionsFinder.html#gammapy.makers.WobbleRegionsFinder

fabiopintore commented:
thanks @maxnoe . I wasn't useing `WobbleRegionsFinder` to avoid the limitations on the overlapping region. However, it's true that Li&Ma cannot be used for correlated counts. A possibility could be to mask overlapping regions before computing the theta2 plot, but I think this is a too complex approach. Therefore, I've implemented the `WobbleRegionsFinder`.

p.s.: sorry, by mistake I removed you from the reviewer and I've realised only now :(

fabiopintore commented:
I'm currently working on the solution we proposed. I would like to do a new push by today, if possible.
So, please do not close it.
